core/src/main/java/io/stargate/core/activator/BaseActivator.java

```diff
@@ -280,15 +280,15 @@ protected void stopService() {
 
     private final Hashtable<String, String> properties;
 
-    public ServiceAndProperties(
-        Object service, Class<?> targetServiceClass, Hashtable<String, String> properties) {
+    public <T, E extends T> ServiceAndProperties(
```

Comment:
do I actually need `E`?

Reply:
Do you mean `T service, Class<T> targetServiceClass` instead of `E service, Class<T>`?

Reply:
Yup..

Reply:
Having a distinction here may be convenient in cases where the same instance is used for multiple services under different interfaces.

Reply:
Since we are not keeping the generics it does not matter much, but I made it clear with single generic param:

`T service, Class<? super T> targetServiceClass`

This is actually the correct form imo.

Reply:
sgtm :+1: